core/capabilities/ccip/ccipevm/msghasher_test.go

```diff
@@ -349,6 +354,97 @@ func TestMessagerHasher_againstRmnSharedVector(t *testing.T) {
 	})
 
+	t.Run("solana vector", func(t *testing.T) {
+		key, err := solanago.NewRandomPrivateKey()
+		require.NoError(t, err)
+
+		// Borsh encoded extra args
+		ea := fee_quoter.EVMExtraArgsV2{
+			GasLimit:                 agbinary.Uint128{Lo: 5000, Hi: 0},
+			AllowOutOfOrderExecution: false,
+		}
+
+		var extraArgsbuf bytes.Buffer
+		encoder := agbinary.NewBorshEncoder(&extraArgsbuf)
+		err = ea.MarshalWithEncoder(encoder)
+		require.NoError(t, err)
+
+		destGasAmount := uint32(10)
+		destExecData := make([]byte, 4)
+		binary.LittleEndian.PutUint32(destExecData, destGasAmount)
+
+		var (
+			// header fields
+			messageID           = mustBytes32FromString(t, "0xcdad95e113e35cf691295c1f42455d41062ba9a1b96a6280c1a5a678ef801721")
+			sourceChainSelector = cciptypes.ChainSelector(124615329519749607)   // solana
+			destChainSelector   = cciptypes.ChainSelector(16015286601757825753) // sepolia
+			sequenceNumber      = cciptypes.SeqNum(386)
+			nonce               = uint64(1)
+			// message fields
+			// sender is parsed unpadded since its emitted unpadded from EVM.
+			senderAddress = cciptypes.UnknownAddress(key.PublicKey().Bytes())
+			// onRampAddress is parsed padded because its set as a padded address in the offRamp
+			onRampAddress = key.PublicKey().Bytes()
+			dataField     = "0x"
+			// receiver address is parsed padded because its emitted as padded from EVM.
+			receiverAddress = cciptypes.UnknownAddress(hexutil.MustDecode("0x000000000000000000000000269895ac2a2ec6e1df37f68acfbbda53e62b71b1"))
+			feeTokenAmount  = big.NewInt(114310554250104)
+			feeValueJuels   = big.NewInt(16499514422603741)
+			tokenAmounts    = []cciptypes.RampTokenAmount{
+				{
+					// parsed unpadded since its emitted unpadded from SVM.
+					SourcePoolAddress: cciptypes.UnknownAddress(key.PublicKey().Bytes()),
+					// parsed padded because its emitted padded from EVM.
+					DestTokenAddress: cciptypes.UnknownAddress(hexutil.MustDecode("0x000000000000000000000000b8d6a6a41d5dd732aec3c438e91523b7613b963b")),
+					// extra data always abi-encoded
+					ExtraData: cciptypes.Bytes(hexutil.MustDecode("0x0000000000000000000000000000000000000000000000000000000000000012")),
+					Amount:    cciptypes.NewBigInt(big.NewInt(100000000000000000)),
+					// dest exec data always abi-encoded
+					DestExecData: destExecData,
+				},
+			}
+
+			msg = cciptypes.Message{
+				Header: cciptypes.RampMessageHeader{
+					MessageID:           messageID,
+					SourceChainSelector: sourceChainSelector,
+					DestChainSelector:   destChainSelector,
+					SequenceNumber:      sequenceNumber,
+					Nonce:               nonce,
+					MsgHash:             cciptypes.Bytes32{},
+					OnRamp:              onRampAddress,
+				},
+				Sender:         senderAddress,
+				Data:           hexutil.MustDecode(dataField),
+				Receiver:       receiverAddress,
+				ExtraArgs:      append(evmExtraArgsV2Tag, extraArgsbuf.Bytes()...),
```

Comment:
The extra args tag for sol -> evm messages is the same as the extra args tag for EVM -> everything except solana? Isn't the point of the tag to make the domain separation clean b/w source and dest?

Reply:
Good catch, shouldn't use this value from evm contract. The reason it works because the SVM [tag](https://github.com/smartcontractkit/chainlink-ccip/blob/1b2ee24da54bddef8f3943dc84102686f2890f87/chains/solana/contracts/programs/ccip-router/src/extra_args.rs#L9) is the same with [EVM](https://github.com/smartcontractkit/chainlink-ccip/blob/1b2ee24da54bddef8f3943dc84102686f2890f87/chains/solana/utils/ccip/ccip_messages.go#L20).

Reply:
Was this a conscious decision or a copy/pasta error?

Reply:
The onchain tag are indeed the same ([svm](https://github.com/smartcontractkit/chainlink-ccip/blob/1b2ee24da54bddef8f3943dc84102686f2890f87/chains/solana/contracts/programs/ccip-router/src/extra_args.rs#L9), [evm](https://github.com/smartcontractkit/chainlink/blob/f2783f8ace87db2c3362a417f28e45d14285e643/contracts/src/v0.8/ccip/libraries/Client.sol#L43))

Reply:
copy/paste error when I write the msgHasher test, should use the one from [here](https://github.com/smartcontractkit/chainlink/blob/79bbad013d4ed9f4cf9a95a37506aa672a9fff90/core/capabilities/ccip/ccipsolana/extradatadecoder.go#L27). But since it's not exported, now I just create a local var in the unit test